examples/surrealdb/src/main.rs

```diff
@@ -0,0 +1,676 @@
+                    .bind(("embeds", two))
+                    .await
+                    .map_err(|e| e.to_string())?;
+                match res.take::<Option<PageContent>>(0) {
+                    Ok(Some(v)) => {
+                        results.push('\n');
+                        results.push_str(&v.title);
+                    }
+                    Ok(None) => return Err("No PageContent found".to_string()),
+                    Err(e) => return Ok(e.to_string()),
+                }
+            }
+            Ok(results)
+        } else {
+            Err(String::from("No documents found to update"))
+        }
+    }
+
+    async fn insert_documents(&self, page_names: String) -> Result<String, String> {
+        // Get each page name separated by a comma and add _ in between words
+        let page_names = page_names
+            .trim()
+            .split(",")
+            .map(|p| p.split_whitespace().collect::<Vec<&str>>().join("_"));
+        let mut result = String::new();
+
+        let results = page_names
+            .map(|page| {
+                std::thread::spawn(move || {
+                    let url = page_for(&page);
+                    let res = ureq::get(url).call();
+                    match res {
+                        Ok(mut o) => o.body_mut().read_to_string().unwrap(),
+                        Err(_) => format!("No page {page} found"),
+                    }
+                })
+            })
+            .collect::<Vec<_>>();
+
+        for res in results {
+            let s = res.join().unwrap();
+            let mut content: PageContent = match serde_json::from_str(&s) {
+                Ok(content) => content,
+                Err(_) => return Err(s),
+            };
+            // Add an underscore again as response from Wikipedia won't have it
+            content.title = content
+                .title
+                .split_whitespace()
+                .collect::<Vec<&str>>()
+                .join("_");
+            result.push('\n');
+            result.push_str(&self.add_document(content).await);
+        }
+        Ok(result)
+    }
+
+    async fn add_document(&self, content: PageContent) -> String {
+        let doc = RecordId::from_table_key("document", &content.title);
+
+        let res = self
+            .db
+            .query("CREATE ONLY $doc SET title = $title, extract = $extract;")
+            .bind(("doc", doc))
+            .bind(("title", content.title))
+            .bind(("extract", content.extract));
+        match res.await {
+            Ok(mut r) => match r.take::<Option<PageContent>>(0) {
+                Ok(Some(good)) => format!("{good:?}"),
+                Ok(None) => "No PageContent found".to_string(),
+                Err(e) => e.to_string(),
+            },
+            Err(e) => e.to_string(),
+        }
+    }
+
+    fn update(&mut self, message: Message) {
+        use Message as M;
+
+        let rt = self.rt.take().unwrap();
+
+        rt.block_on(async {
```

Comment:
I think the desired way to do this in iced is return iced::Task by update function, which passes those Tasks to the async iced runtime. Those can itself generate messages on-complete instead of blocking inside the update function. See  for the Task example:

https://github.com/iced-rs/iced/blob/d475ae5b454b82cb549d6dd961ae158eb5ac7f19/examples/pokedex/src/main.rs#L43